apps/stats-dapp/src/provider/hooks/useProposals.ts

```diff
@@ -605,3 +607,106 @@ export function useVotes(votesReqQuery: VotesQuery): VotesPage {
 
   return votes;
 }
+
+export type TimeRange =
+  | 'all'
+  | 'one-year'
+  | 'six-months'
+  | 'three-months'
+  | 'year-to-date';
+
+export type ProposalsOvertimeTotalCount = {
+  month: number;
+  proposalType: ProposalType | string;
+  totalCount: number;
+}[];
+
+/**
+ * Proposals Overtime Count
+ * @return ProposalsOvertimeTotalCount - Proposal overtime total count data
+ * */
+export function useProposalsOvertimeTotalCount(
+  timeRange: TimeRange
+): Loadable<ProposalsOvertimeTotalCount> {
+  const [proposalsOvertimeCount, setProposalsOvertimeCount] = useState<
+    Loadable<ProposalsOvertimeTotalCount>
+  >({
+    isLoading: true,
+    val: null,
+    isFailed: false,
+  });
+
+  const client = useApolloClient();
+
+  const {
+    blockTime,
+    metaData: { lastProcessBlock },
+  } = useStatsContext();
+
+  const blockRanges: BlockRanges = useMemo(() => {
+    let _blockRanges: BlockRanges = [];
+
+    switch (timeRange) {
+      case 'all':
+        _blockRanges = getBlockRanges(12, lastProcessBlock);
+        break;
+      case 'three-months':
+        _blockRanges = getBlockRanges(3, lastProcessBlock);
+        break;
+      case 'six-months':
+        _blockRanges = getBlockRanges(6, lastProcessBlock);
+        break;
+      case 'one-year':
+        _blockRanges = getBlockRanges(12, lastProcessBlock);
+        break;
+      default:
+        _blockRanges = getBlockRanges(12, lastProcessBlock);
+    }
+
+    return _blockRanges;
+  }, [timeRange, lastProcessBlock, blockTime]);
+
+  useEffect(() => {
+    Promise.all(
+      blockRanges.map((parameter) => {
+        return client.query({
+          query: ProposalsOvertimeTotalCountDocument,
+          variables: parameter,
+        });
+      })
+    )
```

Comment:
Before doing the mapping from `parameters` to `query` you should set the state to be `isLoading= true`